include/tc/lang/parser.h

```diff
@@ -149,6 +149,24 @@ struct Parser {
     auto r = parseExp();
     return RangeConstraint::create(id->range(), id, l, r);
   }
+  TreeRef parseLetBinding() {
+    auto ident = parseIdent();
+    L.expect('=');
+    auto exp = parseExp();
+    return Let::create(ident->range(), ident, exp);
+  }
+  TreeRef parseWhereClause() {
+    auto lookahead = L.lookahead();
+    if (lookahead.kind == '=') {
+      return parseLetBinding();
+    } else if (lookahead.kind == TK_IN) {
+      return parseRangeConstraint();
+    } else {
+      auto exp = parseExp();
+      L.expect(TK_EXISTS);
+      return Exists::create(exp->range(), {exp});
```

Comment:
This change is not equivalent, because it changes the parsing from `<exp> exists` to `exists <exp>`. This is generally why the parser code does not put `parseExp()` into the `::create` statements.